data_diff/diff_tables.py

```diff
@@ -73,11 +76,23 @@ def __post_init__(self):
 
     @property
     def _key_column(self):
-        return ColumnName(self.key_column)
+        return self._quote_column(self.key_column)
 
     @property
     def _update_column(self):
-        return ColumnName(self.update_column)
+        return self._quote_column(self.update_column)
+
+    def _quote_column(self, c):
+        if self.quote_columns:
+            return self.database.quote(c)
+        return c
+
+    def with_schema(self) -> "TableSegment":
```

Comment:
For a snowflake table (see above error) this is somehow empty. I think then we'll need to error early

Reply:
Ahhh it's because it's of a string type (I'm testing something else haha)